fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/rdf/impl/VersionsRdfContext.java

```diff
@@ -66,44 +62,27 @@
      *
      * @param resource the resource
      * @param idTranslator the id translator
-     * @throws RepositoryException if repository exception occurred
      */
     public VersionsRdfContext(final FedoraResource resource,
-                              final IdentifierConverter<Resource, FedoraResource> idTranslator)
-        throws RepositoryException {
+                              final IdentifierConverter<Resource, FedoraResource> idTranslator) {
         super(idTranslator.reverse().convert(resource).asNode());
-        this.versionHistory = resource.getVersionHistory();
+        this.resource = resource;
         concat(versionTriples());
     }
 
     @SuppressWarnings("unchecked")
-    private Stream<Triple> versionTriples() throws RepositoryException {
-        return iteratorToStream(versionHistory.getAllVersions())
-            /* Discard jcr:rootVersion */
-            .filter(uncheck((final Version v) -> !v.getName().equals(versionHistory.getRootVersion().getName())))
-            /* Omit unlabelled versions */
-            .filter(uncheck((final Version v) -> {
-                final String[] labels = versionHistory.getVersionLabels(v);
-                if (labels.length == 0) {
-                    LOGGER.warn("An unlabeled version for {} was found!  Omitting from version listing!",
-                        topic().getURI());
-                } else if (labels.length > 1) {
-                    LOGGER.info("Multiple version labels found for {}!  Using first label, \"{}\".", topic().getURI());
-                }
-                return labels.length > 0;
-            }))
-            .flatMap(UncheckedFunction.uncheck((final Version v) -> {
-                final String[] labels = versionHistory.getVersionLabels(v);
+    private Stream<Triple> versionTriples() {
+        return resource.getVersionLabels()
+            .flatMap(label -> {
                 final Node versionSubject
-                        = createProperty(topic() + "/" + FCR_VERSIONS + "/" + labels[0]).asNode();
-
-                return Stream.concat(
-                        Arrays.stream(labels).map(x -> create(versionSubject, HAS_VERSION_LABEL.asNode(),
-                                createLiteral(x))),
-
-                        Stream.of(create(topic(), HAS_VERSION.asNode(), versionSubject),
-                            create(versionSubject, CREATED_DATE.asNode(),
-                                createTypedLiteral(v.getCreated()).asNode())));
-            }));
+                        = createProperty(topic() + "/" + FCR_VERSIONS + "/" + label).asNode();
+                // TODO - convert this to java.time.* classes once the kernel-api interface changes
+                final Calendar cal = getInstance();
```

Comment:
It's up to you, whether to spend any more time with this or just leave it be since it won't be around for that long anyway.